PCI PT Hotplug/Hotunplug fixes for arch ppc64

TasmiyaNalatwad opened the pull request:
Added 2 new test cases, where hitplog unplug is performed for multiple times and a random reboot is done between hotplug and unplug and the device avialability is checked.

Commit:
multivm-stress:Update script to test all edgecases

This patch captures multiple edge cases to test multivm scenarios. The following updates are added:

add stress_time parameter to run stress test for n seconds before starting stress_events
add debug_dir parameter to save the the debug files
add dump_options parameter to specify virsh dump type
update guest on_crash value to preserve in case of crash
add function check_call_traces to check for any call trace in dmesg
during stress, check for guest state and call traces every ten minutes
if any crashed vms, dump the vm to the debug_dir for further analysis
run stress_events in the remaining stable vms if present, else skip
check for error messages and fail the test if found

Signed-off-by: Misbah Anjum N <misanjum@linux.vnet.ibm.com>

TasmiyaNalatwad committed:
PCI PT Hotplug/Hotunplug fixes for arch ppc64
Added 2 new test cases, where hitplog unplug is performed for multiple times and a random reboot is done between hotplug and unplug and the device avialability is checked.

Signed-off-by: Tasmiya Nalatwad <tasmiya@linux.vnet.ibm.com>

misanjumn commented:
@TasmiyaNalatwad 
Can you run pylint and tackle the `E` errors and other commons errors like unsed parameters, etc.
Thank you.

TasmiyaNalatwad commented:
@misanjumn I had already executed pycodestyle on the PR to check syntax/indentation errors.

What pylint is showing is from older code, not from what was added by me. So keeping those things untouched as it might not be accepted in upstream to make those changes

TasmiyaNalatwad committed:
Added EEH test cases for PCI PT NIC devices
Added 2 test cases
1. Inject EEH from host console for pci pt device
2. Inject EEH from guest console for pci pt device ---> This feature is under developement, Patches are under review. I will enable the test once the feture is in production.

Signed-off-by: Tasmiya Nalatwad <tasmiya@linux.vnet.ibm.com>
